embeddings/data/datamodule.py

```diff
@@ -107,47 +103,77 @@ def prepare_data(self) -> None:
     def setup(self, stage: Optional[str] = None) -> None:
         self.dataset = self.load_dataset()
         self.downsample_dataset()
-        self.num_classes = self.get_num_classes()
+        self.prepare_labels()
         self.process_data()
 
+    @abc.abstractmethod
+    def prepare_labels(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def _class_encode_column(self, column_name: str) -> None:
+        pass
+
     def process_data(self) -> None:
         columns = [c for c in self.dataset["train"].column_names if c not in self.LOADER_COLUMNS]
         self.dataset = self.dataset.map(
             self.convert_to_features,
             batched=True,
+            batch_size=self.processing_batch_size,
             remove_columns=columns,
         )
+        self._class_encode_column("labels")
         self.dataset.set_format(type="torch")
 
     def train_dataloader(self) -> DataLoader[HuggingFaceDataset]:
-        return DataLoader(self.dataset["train"], batch_size=self.train_batch_size)  # type: ignore
+        return DataLoader(
+            dataset=self.dataset["train"],  # type: ignore
+            batch_size=self.train_batch_size,
+            collate_fn=self.collate_fn,
+        )
 
     # Ignoring the type of val_dataloader method from supertype "DataHooks" allowing for None
     # and training without validation dataset.
     def val_dataloader(self) -> Optional[DataLoader[HuggingFaceDataset]]:  # type: ignore
         if "validation" in self.dataset:
-            return DataLoader(self.dataset["validation"], batch_size=self.eval_batch_size)  # type: ignore
+            return DataLoader(
+                dataset=self.dataset["validation"],  # type: ignore
+                batch_size=self.eval_batch_size,
+                collate_fn=self.collate_fn,
+            )
         else:
             return None
 
     def test_dataloader(self) -> DataLoader[HuggingFaceDataset]:
-        return DataLoader(self.dataset["test"], batch_size=self.eval_batch_size)  # type: ignore
+        return DataLoader(
+            dataset=self.dataset["test"],  # type: ignore
+            batch_size=self.eval_batch_size,
+            collate_fn=self.collate_fn,
+        )
 
     @abc.abstractmethod
     def convert_to_features(
         self, example_batch: Dict[str, Any], indices: Optional[List[int]] = None
     ) -> BatchEncoding:
         pass
 
+    @property
+    def collate_fn(self) -> Optional[Callable[[Any], Any]]:
+        return None
+
 
 class TextClassificationDataModule(HuggingFaceDataModule):
+    DEFAULT_BATCH_ENCODING_KWARGS = {
+        "padding": True,
+        "truncation": True,
+    }
+
     def __init__(
         self,
         tokenizer_name_or_path: str,
         dataset_name: str,
         text_fields: Union[str, Sequence[str]],
         target_field: str,
-        max_seq_length: Optional[int] = None,
```

Comment:
I did not want to double the parameters with HuggingFaceDataModule and you could add this (and other) argument to HuggingFaceDataModule through keyword arguments. Do you think it will be confusing to the user? If so I can change it back the way it was previosuly.

Reply:
I am not a fan of passing args through kwargs, but I see there are many of them, so ok.